### DISCUSSION TITLE
Bearer Token Not Working?

### DISCUSSION THREAD

**Discussion by tiagoapp** [adonisjs/core#3949] [Help] [answered]
I'm new to Adonis, after several years of Laravel usage, and was trying to get a basic api authenticated route. 

But for some reason keep getting unauthorized access?

```ts 
// post /login

const token = await auth.use('api').attempt(email, password, {
     expiresIn: '10 days'
})
return token
          
// all is good, i can get the token
```

```ts
// routes.ts

Route.group(() => {
  Route.get('/all', 'ArticlesController.all')
}).prefix('api/v1/articles').middleware('auth');
```

Making a get request via Insomnia with Authorization: Bearer (Token) always returns "E_UNAUTHORIZED_ACCESS: Unauthorized access"

PS: I've seen screenshots with "pretty" error screen pages... is there any setting to enable those or is it some package?

**Comment by thetutlage** [comment, marked as answer]
Can you share the headers Insomnia is sending?

**Reply by tiagoapp** [reply]
Sure

```
> GET /api/v1/articles/all HTTP/1.1
> Host: 127.0.0.1:3333
> User-Agent: insomnia/2022.6.0
> Cookie: cl9pegpbo0000rrvs4b0w4aal=e%3AuEhCBNNIaaYIEihKvm1GohF7Lo7Zps46BPcBEgaEEYRkCOUtAN_-EaLo2HnROpMmNNDah6Wgem8u1S7LGM5hFw.SDV5dnRnZ2pNcjk0dXowdw.5bm_-bCD2eZcLwxUmN57KfftKG8E3sB8Xuj571Fhu5c; adonis-session=s%3AeyJtZXNzYWdlIjoiY2w5cGVncGJvMDAwMHJydnM0YjB3NGFhbCIsInB1cnBvc2UiOiJhZG9uaXMtc2Vzc2lvbiJ9.62hAqYKvGyQEq8Dy1WN03a6HbVXPPqo9AvJZWFNH1Pw
> Authorization: Bearer MTE.ht3Xvg6NTOI7IL8kcmeCh82jmdUa9SmgXPJdqMYSf_t0PoFFsyw2jGyeL_ZQ
> Accept: */*
```

**Reply by thetutlage** [reply]
Ohhh I see. You are using the auth middleware, but not telling it which guard to use. It should be as follows.

```ts
middleware('auth:api')
```

**Reply by tiagoapp** [reply]
Got it, thanks for the super fast reply! Pretty simple actually. Works now.
